src/crypto/sym.rs

```diff
@@ -164,6 +162,21 @@ impl SymmetricKeyAlgorithm {
     /// Uses an IV of all zeroes, as specified in the openpgp cfb mode.
     /// Does not do resynchronization.
     pub fn decrypt_protected<'a>(self, key: &[u8], ciphertext: &'a mut [u8]) -> Result<&'a [u8]> {
+        #[inline]
+        fn calculate_sha1_unchecked<I, T>(data: I) -> Vec<u8>
+        where
+            T: AsRef<[u8]>,
+            I: IntoIterator<Item = T>,
+        {
+            use sha1::{Digest, Sha1};
+
+            let mut digest = Sha1::new();
+            for chunk in data {
+                digest.update(chunk.as_ref());
+            }
+            digest.finalize()[..20].to_vec()
```

Comment:
you can return an array here, no need to allocate

Reply:
right, cool